docs: clarify internal endpoint access (#7219)
grafana/tempo: docs/sources/tempo/operations/authentication.md

It's assumed that clients setting `X-Scope-OrgID` are trusted clients, and the responsibility of populating this value should be handled by the authenticating reverse proxy.
For more information, read the [multi-tenancy](https://grafana.com/docs/tempo/<TEMPO_VERSION>/operations/manage-advanced-systems/multitenancy/) documentation.
{{< /admonition >}}

## Protect internal and administrative endpoints

Tempo exposes operational endpoints on component HTTP and gRPC listeners for status pages, ring inspection, flush, shutdown, downscale workflows, and backend scheduling.
These endpoints are intended for internal operator access, automation, and component-to-component communication.

> They are not tenant-scoped API endpoints and should not be exposed to tenants or untrusted clients.

Reviewer: ```suggestion
They aren't tenant-scoped API endpoints and shouldn't be exposed to tenants or untrusted clients.
```

Expose only the tenant-facing API routes that are required for ingestion and querying through public or tenant-facing ingress.
Keep component listeners, such as backend-scheduler and live-store, restricted to internal networks.

In monolithic deployments, or any deployment where a reverse proxy forwards a component listener, configure the proxy to block internal routes before forwarding requests to Tempo.
Depending on the enabled components, internal HTTP routes include:

- `/flush`
- `/shutdown`
- `/partition-ring`
- `/live-store/prepare-partition-downscale`
- `/live-store/prepare-downscale`

> - `/status/backendscheduler`

Reviewer 2: It might be easier to list the endpoints that are user-facing instead.

Reviewer 3: +1, can add the /overrides endpoints as well.

The backend-scheduler also exposes gRPC methods for backend workers and redaction workflows.

> Do not expose the backend-scheduler gRPC service to untrusted clients.

Reviewer: Minor stylistic thing: We use contractions in our docs.
```suggestion
Don't expose the backend-scheduler gRPC service to untrusted clients.
```

Restrict access to trusted backend workers and operators, including methods such as:

- `/tempopb.BackendScheduler/Next`
- `/tempopb.BackendScheduler/UpdateJob`
- `/tempopb.BackendScheduler/SubmitRedaction`

The `X-Scope-OrgID` header scopes tenant data.

> It is not an administrative authorization mechanism for cluster-wide operational endpoints.

Reviewer: ```suggestion
It isn't an administrative authorization mechanism for cluster-wide operational endpoints.
```
